### ISSUE TITLE
QNAP and new SR repo

### ISSUE THREAD

**Issue by dltyih** [SickRage/sickrage-issues#297]
I have been using SickRageTV and wanted to switch over to the new repo. I have installed this from the old repo on a QNP TVS-871 and using the default QNAP App Store Python (version 2.7.3). 

QNAP App Store does not have a more up to date version of the Python. I read somewhere that a newer Python (2.7.10) is needed but QNAP does not make it. I have found that someone has made a Qpython 2.7.10 but after installing it, this 2.7.10 seems to co-exist with the Python 2.7.3 on my NAS, and the SR was still using the old version. 

Is there a guide on how to set up the new repo SR on a QNAP? I found that wiki talked about a QNAP guide that is out of date.

Thank you!

**Comment by SickRage2** [comment]
Thanks for the issue report! Before a real human comes by, please make sure your report has all the below criteria checked

- [ ] Include basic information: Branch/Commit, OS, What you did, What happened, What you expected
- [ ] Enable debug logging (be sure to disable after the bug is fixed)
- [ ] Post debug logs, either inline (for smaller logs) or using gist)

Please make sure you also read [how to create an issue](https://github.com/SickRage/sickrage-issues/blob/master/README.md) and followed all of the steps. 

The title should describe your issue. Having "SR not working" or "I get this bug" for 100 issues, isn't really helpful. We will close issues if there isn't enough information.

Sometimes the devs may seem like grunts and respond with short answers. This isn't (always) because the dev hates you, but because he's on mobile or busy fixing bugs. If something isn't clear, please let us know, and this bot may get updated to automatically answer you.

Thanks!

**Comment by neoatomic** [comment]
That QNAP is a ARM based device correct.?
You should be able to install this package. : http://apps.qnap.community/11-community/5-sickrage

However there are 2 major problems with it. 

* Still points to the old Repository. (that you can solve by manual git commands)
* SSL is broken. (bad python package)

Sadly all the how-to's/fixes etc are not available anymore as some1 found is necessary to close the old issue tracker with all that data... 

But from what i recall, you need to use this modified python package. : 
https://www.dropbox.com/s/7udod90304eh0jt/Python_2.7.10_arm-x19_by_ricebus.qpkg?dl=0

In this topic there are some QNAP users, maybe they have some answers for you.  https://github.com/SickRage/sickrage-issues/issues/6

Also checked the QNAP forums.? I'm sure your re not the only person with this problem.

**Comment by neoatomic** [comment]
@dltyih This method should also work for you. However its not a solution for the SSL problem. That has to do with the Python version in the package...

https://github.com/SickRage/sickrage-issues/wiki/Sickrage-installation-packages#switching-the-dirty-way

**Comment by dltyih** [comment]
Unfortunately it is an Intel i7  device, and I am using the SR package that you referred. I did not switch to the new repo because the Python on my nas is still 2.7.3. If only I can find 2.7.10 .....

I have found a modified Python 2.7.10 packed for the QNAP (from memory I think it is called Qpython) but the SR I installed does not know to make use of that. 





On 23 Dec 2015, at 5:07 AM, neoatomic <notifications@github.com> wrote:

That QNAP is a ARM based device correct.?
You should be able to install this package. : http://apps.qnap.community/11-community/5-sickrage

However there are 2 major problems with it.

Still points to the old Repository. (that you can solve by manual git commands)
SSL is broken. (bad python package)
Sadly all the how-to's/fixes etc are not available anymore as some1 found is necessary to close the old issue tracker with all that data...

But from what i recall, you need to use this modified python package. : 
https://www.dropbox.com/s/7udod90304eh0jt/Python_2.7.10_arm-x19_by_ricebus.qpkg?dl=0

In this topic there are some QNAP users, maybe they have some answers for you. #6

Also checked the QNAP forums.? I'm sure your re not the only person with this problem.

**Comment by neoatomic** [comment]
If its a Intel then you need to use this package. :
http://apps.qnap.community/11-community/17-qsickrage
That has working python/SSL. Only need to do the git commands to switch to the new repo than.

**Comment by dltyih** [comment]
Thank you for your help so far, neoatomic. Much appreciated.

I need a little more help, if I may. I installed QSickRage as you have pointed out, on my QNAP. I entered ssh to my QNAP using OSX Terminal (ssh -l admin 192.xxx.xxx.xxx), and changed directory to the qpkg folder. Typed in "git remote set-url origin https://github.com/SickRage/SickRage.git", and got the result:

fatal: Not a git repository (or any parent up to mount point /share/CACHEDEV1_DATA)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).

Is there somewhere that I can look this thing up how to resolve it?

Many thanks!

**Comment by miigotu** [comment]
First, http://apps.qnap.community/ has Python 2.7.10 for arm and 2.7.9 for intel.
Second, you should be using sickbeard-tvrage
Back up your db and settings (through the webui in SR, or copy sickbeard.db and config.ini)
Then remove QSickRage and install Sickbeard-TVRage by @clinton-hall
```bash
cd /share/CACHEDEV1_DATA/.qpkg/Sickbeard-TVRage/
sed -i 's/SiCKRAGETV/SickRage/g' *.sh
cd Sickbeard-TVRage
git remote set-url origin https://github.com/SickRage/SickRage.git
git fetch origin
git prune && git remote prune origin
git reset --hard origin/master
```

**Comment by dltyih** [comment]
Thank you, miigotu!

I am a little puzzled though. The Python 2.7.9 in the QNAP apps community is one done by Stephane called QPython. It does install but Clinton Hall's SickRage package looks for the one without the "Q". I was searching for resources for the SR package to use QPython, but wasn't quite sure what files to edit. Do you know? Thanks in advance

**Comment by miigotu** [comment]
Clinton halls package needs updated still. Waiting on him

**Comment by clinton-hall** [comment]
Try this package https://www.dropbox.com/s/j1svazqdi9ieq82/SickBeard-TVRage_151227.qpkg

This now allows a few things... 

1. It searches for the latest version of Python installed.
2. It actually opens a webpage asking you which version of SickRage to install... this gives the option for the 2 "main" repositories and also allows you to specify a custom url.

After specifying which fork to install, you need to restart the App and this will then download the chosen version.

Currently I do observe some issues on some platforms as there are a bunch of "broken" combinations of Python qpkgs (that don't have some needed modules) and SickRage Forks/Branches.

For what it is worth... I also have the Asustor apkg that also allows the user to select which fork to install. As far as I can tell, all combinations appear to work.
https://www.dropbox.com/s/f016hij9ebw28qd/sickbeard-tvrage_20151227_any.apk

**Comment by clinton-hall** [comment]
issue with missing modules currently seen on QNAP...

```
Starting SickBeard-TVRage
Traceback (most recent call last):
  File "SickBeard.py", line 66, in <module>
    import sickbeard
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/sickbeard/__init__.py", line 36, in <module>
    from sickbeard import metadata
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/sickbeard/metadata/__init__.py", line 22, in <module>
    from sickbeard.metadata import kodi, kodi_12plus, mediabrowser, ps3, wdtv, tivo, mede8er, generic, helpers
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/sickbeard/metadata/kodi.py", line 23, in <module>
    from sickbeard.metadata import generic
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/sickbeard/metadata/generic.py", line 32, in <module>
    from sickbeard import helpers
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/sickbeard/helpers.py", line 45, in <module>
    import adba
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/adba/__init__.py", line 24, in <module>
    from aniDBAbstracter import Anime, Episode
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/adba/aniDBAbstracter.py", line 23, in <module>
    import aniDBfileInfo as fileInfo
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/adba/aniDBfileInfo.py", line 30, in <module>
    import requests
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/requests/__init__.py", line 53, in <module>
    from .packages.urllib3.contrib import pyopenssl
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/requests/packages/urllib3/contrib/pyopenssl.py", line 48, in <module>
    from ndg.httpsclient.ssl_peer_verification import SUBJ_ALT_NAME_SUPPORT
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/ndg/__init__.py", line 19, in <module>
    __import__('pkg_resources').declare_namespace(__name__)
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/pkg_resources.py", line 2627, in <module>
    add_activation_listener(lambda dist: dist.activate())
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/pkg_resources.py", line 683, in subscribe
    callback(dist)
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/pkg_resources.py", line 2627, in <lambda>
    add_activation_listener(lambda dist: dist.activate())
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/pkg_resources.py", line 2176, in activate
    map(declare_namespace, self._get_metadata('namespace_packages.txt'))
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/pkg_resources.py", line 1794, in declare_namespace
    _handle_ns(packageName, path_item)
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/pkg_resources.py", line 1765, in _handle_ns
    loader.load_module(packageName); module.__path__ = path
  File "/opt/QPython2/lib/python2.7/pkgutil.py", line 246, in load_module
    mod = imp.load_module(fullname, self.file, self.filename, self.etc)
  File "/share/MD0_DATA/.qpkg/QPython2/lib/python2.7/site-packages/zopeskel/__init__.py", line 2, in <module>
    from zopeskel.basic_namespace import BasicNamespace
  File "/share/MD0_DATA/.qpkg/QPython2/lib/python2.7/site-packages/zopeskel/basic_namespace.py", line 2, in <module>
    from zopeskel.vars import var, DottedVar, StringVar, BooleanVar, TextVar
  File "/share/MD0_DATA/.qpkg/QPython2/lib/python2.7/site-packages/zopeskel/vars.py", line 2, in <module>
    from paste.script.templates import var as base_var
  File "/opt/QPython2/lib/python2.7/site-packages/paste/script/templates.py", line 8, in <module>
    from . import command
  File "/opt/QPython2/lib/python2.7/site-packages/paste/script/command.py", line 54, in <module>
    dist = pkg_resources.get_distribution('PasteScript')
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/pkg_resources.py", line 315, in get_distribution
    if isinstance(dist,Requirement): dist = get_provider(dist)
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/pkg_resources.py", line 197, in get_provider
    return working_set.find(moduleOrReq) or require(str(moduleOrReq))[0]
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/pkg_resources.py", line 670, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/pkg_resources.py", line 569, in resolve
    raise DistributionNotFound(req)  # XXX put more info here
pkg_resources.DistributionNotFound: PasteScript
```

**Comment by alkar** [comment]
@clinton-hall which python version do you use / recommend using?

**Comment by clinton-hall** [comment]
You can email my any suggestions to fix.

I personally use my Asustor NAS for SickRage. I built this based on discussions that included modules in the app were no longer needed when using newer version of Python, but the issue here is the newer version of Python has different included modules for each NAS model!

**Comment by LennyM8472** [comment]
Hi,

I have exactly the same problem as Clinton Hall after trying to add ssl to my setup (which by the way is now completely broken.... after messing around with different python binaries and git commands...)

@ Clinton: Can you add a manual python path input to your setup screen? If I run sickrage via ssh I see that your installation procedure finds 4 python paths in my setup, I know that the normal Python QPKG of QNAP (in the appstore) in combination with the your old setup works (except ssl).

Maybe that is a way to get things partially running again.

thanks
Lenny

**Comment by duramato** [comment]
@LennyM8472 i dont know what versions are on the appstore but ideally for working ssl you need version>2.7.10 atleast or for version<2.7.10 you will need pyOpenSSL

**Comment by LennyM8472** [comment]
Hi,

Python version which is found as default in system:
[~] # python --version
Python 2.7.10

Running Clinton's Script via SSH: From what I see the setup selects the wrong Python version to work correctly (QPython2 somehow won't work and results in the same error as stated from Clinto a couple of posts on top)

[/share/MD0_DATA/.qpkg/SickBeard-TVRage] # sh sickbeard-tvrage.sh start
x86_64
SickBeard-TVRage prestartup checks...
 Checking for git...  Found!
Looking for /usr/bin/python2.7
Looking for /usr/local/bin/python2.7
Looking for /opt/bin/python2.7
Looking for /Apps/opt/bin/python2.7
Looking for /opt/QPython2/bin/python2.7
Found Python Version 2.7.11 at /opt/QPython2/bin/python2.7

Lenny

**Comment by LennyM8472** [comment]
ok, I seem to solve the problem partially atleast:

I changed within the sickbeard-tvrage.sh following code to my setup (/opt/bin/python2.7) - basically forcing the setup to only select my python path as an option (I'm using following python in this case http://forum.qnap.com/viewtopic.php?f=217&t=33567). Even ssl now works now - don't ask why.

ConfigPython(){ #checks if the daemon exists and will link /usr/bin/python to it
	#python dependency checking : /usr/bin/python2.7 /usr/local/bin/python2.7 /opt/bin/python2.7 /Apps/opt/bin/python2.7 /opt/QPython2/bin/python2.7
	VER=0
	DAEMON="None"
	for DAEMON2 in /opt/bin/python2.7

**Comment by mama31** [comment]
I've got this issue when launching sickrage by SSH:

x86_64
SickBeard-TVRage prestartup checks...
 Checking for git...  Found!
Looking for /usr/bin/python2.7
Looking for /usr/local/bin/python2.7
Looking for /opt/bin/python2.7
Looking for /Apps/opt/bin/python2.7
Looking for /opt/QPython2/bin/python2.7
Found Python Version 2.7.5 at /usr/local/bin/python2.7
[/share/.qpkg/SickBeard-TVRage] # Traceback (most recent call last):
  File "server.py", line 17, in <module>
    import SimpleHTTPServer
ImportError: No module named SimpleHTTPServer

Any clue?

**Comment by clinton-hall** [comment]
can you update to the latest QPython version from the community forum?

this is the main issue I struggle with... so many versions of pythin, all with different modules.

**Comment by xenadi** [comment]
using this Python package from the community app store http://apps.qnap.community/11-community/20-python-community-2

i have modified sickbeard-tvrage.sh to search in /share/MD0_DATA/.qpkg/Python/bin/python
and version 2.7.10 is found (the correct version).

p.s. completely new to bash scripting and linux so i have no idea if i shouldn't be directly using the apps in the .qpkg directory.

**Comment by LennyM8472** [comment]
@mama31 modify the sickbeard-tvrage.sh as I mentioned a couple of posts earlier.
That should use the correct version of python to work properly.

**Comment by jverdeyen** [comment]
I know this issue is closed, but just wanted to add my experience.

I've installed `https://dl.dropboxusercontent.com/content_link/bESsueQVkUKfPNiGDWGe54H2FdT0fXUAu1YYUOv44lR4Xh6OIS55QrDyoYzRwvpi/file?dl=1` (`SickBeard-TVRage_151227.qpkg`)

Then visited:
`http://qnap-nas-ip-or-hostname:7073`

Clicked on `sickrage` under `Click here to install from http://github.com/sickrage/sickrage`.

Logged in with ssh onto my Qnap.
`cd /share/MD0_DATA/.qpkg/SickBeard-TVRage`
`vim sickbeard-tvrage.sh`

Replaced following line:
`L48 for DAEMON2 in /opt/bin/python2.7 /usr/bin/python2.7 /usr/local/bin/python2.7 /opt/bin/python2.7 /Apps/opt/bin/python2.7 /opt/QPython2/bin/python2.7`
by
`for DAEMON2 in /opt/bin/python2.7`

My Python version:
```
[/share/MD0_DATA/.qpkg/SickBeard-TVRage] # /opt/bin/python2.7 --version
Python 2.7.11
```

After this all was running fine!
